### ISSUE TITLE
Changing endpoints

### ISSUE THREAD

**Issue by Sladewill** [thephpleague/omnipay#283]
It would be nice to be able to change the endpoints for all of the packages in the case of one of the gateways changes the API url and someone doesn't have the time or knows that they have been changed, with this the end users can update their system and run it seamless until someone is able to update if you even update it.

@maclof

**Comment by judgej** [comment]
And example is Authorize.Net switching over to new servers, and giving out a temporary URL to use over the next few months until the original URL becomes more stable. I like the idea.

Maybe URLs can be handled (defined, constructed) through an object class just for URLs, and the means of modifying the URLs could work through that object. Not a well-thought-through idea, but something less procedural than a bunch of `setWhateverURL` methods could work. If it is based on something like https://github.com/php-fig/http-message/blob/master/src/UriInterface.php then it may even work well with the [PSR-7](http://www.php-fig.org/psr/psr-7/) ideas for OmniPay v3?

**Comment by kayladnls** [comment]
@judgej I do like the idea of a Url or EndPoint object that can be constructed and set for various things. That would be a good thing to add for V3

**Comment by mediabeastnz** [comment]
Yeah agreed this would be helpful as a lot of the providers have different endpoints for UAT etc.
